File: webmvp/src/lib/keyUtils.test.ts
import { describe, expect, it } from "vitest";

import { isKey, transposeKeyBy } from "@/lib/keyUtils";

describe("keyUtils", () => {
  it("transposes keys by semitones", () => {
    expect(transposeKeyBy("C", 2)).toBe("D");
    expect(transposeKeyBy("B", 1)).toBe("C");
    expect(transposeKeyBy("C", -1)).toBe("B");
  });

  it("validates keys", () => {
    expect(isKey("G")).toBe(true);
    expect(isKey("H")).toBe(false);
  });
});

File: mobile/test/domain/key_utils_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:lf_chords/domain/key_utils.dart';

void main() {
  test('transposes keys by semitones', () {
    expect(transposeKeyBy('C', 2), 'D');
    expect(transposeKeyBy('B', 1), 'C');
    expect(transposeKeyBy('C', -1), 'B');
  });

  test('validates keys', () {
    expect(isKey('G'), isTrue);
    expect(isKey('H'), isFalse);
  });
}
